nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/main/java/org/apache/nifi/util/db/JdbcCommon.java

```diff
@@ -711,18 +711,59 @@ public static void setParameters(final PreparedStatement stmt, final Map<String,
         }
     }
 
+    /**
+     * Sets all of the appropriate parameters on the given PreparedStatement, based on the given FlowFile attributes.
+     *
+     * @param stmt the statement to set the parameters on
+     * @param attributes the attributes from which to derive parameter indices, values, and types
+     * @throws SQLException if the PreparedStatement throws a SQLException when the appropriate setter is called
+     */
+    public static void setSensitiveParameters(final PreparedStatement stmt, final Map<String, SensitiveValueWrapper> attributes) throws SQLException {
+        for (final Map.Entry<String, SensitiveValueWrapper> entry : attributes.entrySet()) {
+            final String key = entry.getKey();
+            final boolean isSensitive = entry.getValue().isSensitive();
+            final String value = entry.getValue().getValue();
+            final String logValue = isSensitive ? "???" : value;
+            final Matcher matcher = SQL_TYPE_ATTRIBUTE_PATTERN.matcher(key);
+            if (matcher.matches()) {
+                final int parameterIndex = Integer.parseInt(matcher.group(1));
+
+                final boolean isNumeric = NUMBER_PATTERN.matcher(value).matches();
+                if (!isNumeric) {
+                    throw new SQLDataException("Value of the " + key + " attribute is '" + logValue + "', which is not a valid JDBC numeral type");
+                }
+
+                final int jdbcType = Integer.parseInt(value);
+                final String valueAttrName = "sql.args." + parameterIndex + ".value";
+                final String parameterValue = attributes.get(valueAttrName).getValue();
+                final String logParameterValue = isSensitive ? "???" : parameterValue;
```

Comment:
Using `???` is not clear, particularly when it comes to SQL statements. Recommend changing the value to `MASKED`:
```suggestion
                final String logParameterValue = isSensitive ? "MASKED" : parameterValue;
```